pkcs8/src/one_asymmetric_key.rs

```diff
@@ -0,0 +1,143 @@
+use der::{BitString, ContextualTo1, Decodable, Encodable, Message};
+
+use crate::{
+    common::{Version, _AttributesStub},
+    AlgorithmIdentifier, Error, Result,
+};
+use core::{convert::TryFrom, fmt};
+
+/// PKCS#8 `OneAsymmetricKey`.
+///
+/// ASN.1 structure containing a [`Version`], an [`AlgorithmIdentifier`], private key
+/// data, and optionally public key data, in an algorithm-specific format.
+///
+/// Described in [RFC 5958 Section 2]:
+///
+/// ```text
+/// OneAsymmetricKey ::= SEQUENCE {
+///     version                   Version,
+///     privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
+///     privateKey                PrivateKey,
+///     attributes            [0] Attributes OPTIONAL,
+///     ...,
+///     [[2: publicKey        [1] PublicKey OPTIONAL ]],
+///     ...
+///   }
+///
+/// Version ::= INTEGER { v1(0), v2(1) } (v1, ..., v2)
+///
+/// PrivateKeyAlgorithmIdentifier ::= AlgorithmIdentifier
+///
+/// PrivateKey ::= OCTET STRING
+///
+/// Attributes ::= SET OF Attribute
+///
+/// PublicKey ::= BIT STRING
+/// ```
+///
+/// [RFC 5958 Section 2]: https://datatracker.ietf.org/doc/html/rfc5958#section-2
+#[derive(Clone)]
+pub struct OneAsymmetricKey<'a> {
```

Comment:
FYI, while I think it's fine splitting this out for now so adding it isn't a breaking change, longer term I think it would make sense to merge it with `PrivateKeyInfo`. The two are effectively the same, it's just `OneAsymmetricKey` can have a `PublicKey`.

That's namely because there are a bunch of other types/traits related to it, and if we kept this separate, all of those would need to be duplicated as well:

- [`PrivateKeyDocument`](https://docs.rs/pkcs8/0.6.0/pkcs8/struct.PrivateKeyDocument.html)
- [`FromPrivateKey`](https://docs.rs/pkcs8/0.6.0/pkcs8/trait.FromPrivateKey.html)
- [`ToPrivateKey`](https://docs.rs/pkcs8/0.6.0/pkcs8/trait.ToPrivateKey.html)

Reply:
Do you want me to `impl` those right now, or is this fine for now?

Reply:
No, I'd prefer not to duplicate all of those and the ultimate resolution can be a breaking change to `PrivateKeyInfo` which incorporates the `OneAsymmetricKey` message as well.